### PR TITLE
feat(HTD-77): add privacy page content

### PR DESCRIPTION
**Pull request by RBoraurito**
## Assets

| before | after |
| ------ | ----- |
|        |       |

## Before Merge Checklist

The following steps were checked before merging a PR

- [x] checked and fixed code style issues
- [x] checked and removed unnecessary/junky code
- [x] manually tested introduced solution
- [ ] added needed tests to automate testing this code in the future

### CODE REVIEW

**Review by nephlin7 — APPROVED** [review]
Great job @RBoraurito 💪🏽

File: src/app/(legal)/privacy-policy/page.tsx [hunk 1]
```diff
@@ -1,38 +1,64 @@
...
+function formatLinks(headings: HeadingGroupWithSubItems[]): ContentWithTableProps['links'] {
+  return headings.map(heading => ({
+    text: heading.text,
+    href: `#${kebabCase(heading.text)}`,
+    subItems: heading.subItems.map(subItem => ({
+      text: subItem.text,
+      href: `#${kebabCase(subItem.text)}`,
+    })),
+  }));
+}
```
Comment (nephlin7) [inline src/app/(legal)/privacy-policy/page.tsx]: @RBoraurito, Won't this logic be the same on all rich text pages? Can we make it reusable?

**Review by nephlin7 — APPROVED** [review]
Please take a look at the comment before merge

File: src/app/(legal)/privacy-policy/page.tsx [hunk 2]
```diff
@@ -1,38 +1,64 @@
...
+
+export default async function PrivacyPolicy() {
+  const filePath = path.join(process.cwd(), 'src', 'app', '(legal)', 'privacy-policy', 'content.mdx');
+  const headings = await extractHeadingsFromMDX(filePath);
+  const groups = groupByLevel(headings);
+  const links = formatLinks(groups);
```
Comment (nephlin7) [inline src/app/(legal)/privacy-policy/page.tsx]: Could we also make a `getPageContent(...)` method that returns everything? So we can also reuse it on other rich pages like this one.

File: src/app/(legal)/privacy-policy/page.tsx (hunk 1 above)
Comment (RBoraurito) [reply src/app/(legal)/privacy-policy/page.tsx]: I think it could be reused, let me move it

**Pull request merged**